libs/blog/shared/feature-seo/src/lib/seo/seo.service.ts

```diff
@@ -0,0 +1,219 @@
+import { inject, Injectable } from '@angular/core';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { Meta, MetaDefinition, Title } from '@angular/platform-browser';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { combineLatest, filter, map, switchMap } from 'rxjs';
+
+import { ArticleDetailsStore } from '@angular-love/blog/articles/data-access';
+import { ArticleSeoData } from '@angular-love/contracts/articles';
+
+import { SEO_META_KEYS, SeoMetaKeys } from './seo-meta-keys';
+import { SeoTitleKeys } from './seo-title-keys';
+
+@Injectable()
+export class SeoService {
+  private readonly _articleDetailsStore = inject(ArticleDetailsStore);
+  private readonly _router = inject(Router);
+  private readonly _activatedRoute = inject(ActivatedRoute);
+  private readonly _title = inject(Title);
+  private readonly _meta = inject(Meta);
+
+  constructor() {
+    this.subscribe();
+  }
+
+  private setGlobalSeo(seoData: ArticleSeoData | undefined): void {
+    this.updateTag(seoData?.og_locale || 'en_US', 'ogLocale');
+
+    this.setMetaDescription(
+      seoData?.description ||
+        'Angular.love - ciekawostki oraz rozwiązania dla średnio-zaawansowanych oraz zaawansowanych developerów Angulara.',
+    );
+
+    this.updateTag(seoData?.og_site_name || 'Angular.love', 'ogSiteName');
+  }
+
+  private setArticleSeo(seoData: ArticleSeoData | undefined): void {
```

Comment:
this method can me removed if we expose `setTitle` and `setMeta` to public.

Reply:
@DDonochVA 
I only left `setMeta()` as public method which is bein called in `ArticleDetailsStore`.

`setTitle` is still being set directly in the `SeoService`, which is listenning to router events. This time it is only looking for `route?.data` - which is not set in Article Details Page. Let me know what you think